### PR TITLE
create bump-dna shell script

### PR DESCRIPTION
**Pull request by peeech**

### CODE REVIEW

File: bump-dna.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/bin/sh
+# Simple shell script for altering hashspace of self hosted DNAs
+# type bump-dna for help
+##
+
```
Comment (Br1ght0ne) [inline bump-dna.sh:5]: If a command (like `systemctl restart holochain-conductor`) fails, we don't want to exit with code 0.
`set -e` makes your script bail early with any non-zero code encountered.

```suggestion
##

set -e

```

File: bump-dna.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+CONFIG_FILE="/var/lib/holochain-conductor/conductor-config.toml"
+DNA_DIR="/var/lib/holochain-conductor/dnas/"
+
+[[ -f $CONFIG_FILE ]] || { echo "$CONFIG_FILE is not a file." >&2;exit 1;}
```
Comment (Br1ght0ne) [inline bump-dna.sh]: Guard conditions should be coupled together, and not appear before function definitions. `Error checks` section seems perfectly suitable.

File: bump-dna.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            gsub(/['"'"'"]/,"") # removes all the " and '
+            if ($1 == "hash") hash=$3
+            if ($1 == "id") id=$3
+            if ($0 == "" && id == my_id) print hash
+        }' $CONFIG_FILE )"
+}
+
+file_from_hash(){
+    local a="\"$1\"" # wrap hash in double quotes
+    echo "$( awk -v my_hash=$a '
```
Comment (Br1ght0ne) [inline bump-dna.sh]: `a` is not a great name. How about `$my_hash`, as in `awk` script?

**Review by Br1ght0ne — COMMENTED** [review]
The script is terse, but nothing actually wrong stands out.
Consider running [`shellcheck`][shellcheck] (for instance, via `nix run nixos.shellcheck -c shellcheck`) on the script and fix most of the easy (e.g. quoting, POSIX `sh` vs `bash`, etc.) problems. UUIDs don't contain spaces, of course, but consistency matters too.

If you have trouble with `shellcheck` because of `if` statements in `awk` scripts, run it on the subset of the file. This command skips the functions using `awk`:

```bash
sed '2,31d' bump-dna.sh | shellcheck -
```


[shellcheck]: https://www.shellcheck.net/

**Comment by alastairong** [comment]
@filalex77 Are you proposing validation of UUID format? Given the use case is that we would bump hashes on the fly during testing, and not everyone is in test is technically savvy, I would prefer if we can bump hashes using arbitrary strings.

Basically I think we need to be able to tell people like Rob and Jarod "hey bump the DNA to "001" or "002"

**Comment by Br1ght0ne** [comment]
@alastairong Not at all. I am just saying that bash requires double quoting to properly work with strings that may contain spaces, and it's best to use double quotes around most, if not *all* of variables in a script, to ensure robustness and consistent styling of the script.

Actually, if you say that people can put arbitrary strings as UUIDs, e.g. "my awesome uuid" (note the spaces), double quoting is *required everywhere*.

**Comment by alastairong** [comment]
Thanks for clarifying!

File: bump-dna.sh (hunk 1 above)
Comment (peeech) [reply bump-dna.sh:5]: great suggestion!

File: bump-dna.sh (hunk 3 above)
Comment (peeech) [reply bump-dna.sh]: ah this is actually obsolete with `gsub(/['"'"'"]/,"")`

**Review by Br1ght0ne — APPROVED** [review]
LGTM

**Pull request merged**